### PR TITLE
Remove terminal renderers

### LINKED ISSUE #69865: Remove deprecated terminal renderer APIs
Terminal renderers are becoming stable in https://github.com/Microsoft/vscode/pull/67923. `TerminalRenderer.terminal` and `Terminal.onDidWriteData` are remaining in proposed to give extensions a chance to migrate over by then.

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #69865

### CODE REVIEW

**Comment by Tyriar** [comment]
@alexr00 I disabled `CustomExecution` by throwing exceptions and changing types to `any`, can you remove the rest in this branch?

@GabeDeBacker are we blocked on remove terminal renderers or are we good to merge this when ready?

**Comment by alexr00** [comment]
Yes, I made an issue for it: https://github.com/microsoft/vscode/issues/78540
@GabeDeBacker has moved to the new API so we should be good to go.

**Comment by alexr00** [comment]
CustomExecution is removed in this branch!

**Comment by Tyriar** [comment]
@alexr00 thanks for the quick turn around

image

😍

File: src/vs/vscode.proposed.d.ts [hunk 1]
```diff
@@ -1202,12 +1063,12 @@ declare module 'vscode' {
 		 *  or '$eslint'. Problem matchers can be contributed by an extension using
 		 *  the `problemMatchers` extension point.
 		 */
-		constructor(taskDefinition: TaskDefinition, scope: WorkspaceFolder | TaskScope.Global | TaskScope.Workspace, name: string, source: string, execution?: ProcessExecution | ShellExecution | CustomExecution | CustomExecution2, problemMatchers?: string | string[]);
+		constructor(taskDefinition: TaskDefinition, scope: WorkspaceFolder | TaskScope.Global | TaskScope.Workspace, name: string, source: string, execution?: ProcessExecution | ShellExecution | CustomExecution2, problemMatchers?: string | string[]);
 
 		/**
 		 * The task's execution engine
 		 */
-		execution2?: ProcessExecution | ShellExecution | CustomExecution | CustomExecution2;
```
Comment (Tyriar) [inline src/vs/vscode.proposed.d.ts]: @alexr00 is this changing to just `execution` before it goes stable?

Comment (alexr00) [reply src/vs/vscode.proposed.d.ts]: Yes. I wanted to do the rename in a separate change.

**Pull request merged**